### PR TITLE
Added A sub-menu to change the types of Font and Font-Styles

### PR DESCRIPTION
**Pull request by BarmenduC**
A Menu has been added into the already existing context menu listing various font-families and font-styles that can be used in the note. This caters to #8 and partially to #9 . A few fixes still need to be carried out (would probably be working upon that shortly) .

### CODE REVIEW

File: DeskNoteView.cpp [hunk 1]
```diff
@@ -24,6 +24,9 @@
 #undef B_TRANSLATION_CONTEXT
 #define B_TRANSLATION_CONTEXT "View"
 
+
```
Comment (humdingerb) [inline DeskNoteView.cpp:24]: Remove stray newlines.

File: DeskNoteView.cpp [hunk 2]
```diff
@@ -131,15 +134,22 @@ DeskNoteView::Draw(BRect rct)
...
-	BAlert* alert;
-	BRect windSize, ourSize;
-	BMessenger* messenger;
-	BFont currentFont;
-	font_family currentFamily;
-	font_style currentStyle;
-	uint16 currentFace;
-	const rgb_color* bckgrnd;
-	ssize_t size;
+	BAlert 				*alert;
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Let's keep the style without tabs and added newlines.

File: DeskNoteView.cpp [hunk 3]
```diff
@@ -131,15 +134,22 @@ DeskNoteView::Draw(BRect rct)
...
+	BRect 				windSize,
+					 	ourSize;
+	BMessenger 			*messenger;
+	BMessage			*message;
+	//BFont 			currentFont;
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Remove commented out code (here and below).

File: DeskNoteView.cpp [hunk 4]
```diff
@@ -186,12 +196,49 @@ DeskNoteView::MessageReceived(BMessage* msg)
 			}
 			break;
 		}
+		// Font type
+		case FONT_FAMILY: {
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Keep to the coding style. Opening bracket in new line.

File: DeskNoteView.cpp [hunk 5]
```diff
@@ -186,12 +196,49 @@ DeskNoteView::MessageReceived(BMessage* msg)
...
+
+			fontFamily = NULL;
+			fontStyle = NULL;
+
+			// Setting the font family
+			msg->FindPointer ("source", &ptr);
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Keep to the coding style. No space before opening brackets.

File: DeskNoteView.cpp [hunk 6]
```diff
@@ -231,6 +278,66 @@ DeskNoteView::MouseDown(BPoint point)
...
+	// Taking the number of font families
+	numFamilies = count_font_families();
+	for (int32 i = 0; i < numFamilies; i++) {
+
+		// Getting the font families
+		if (get_font_family(i, &family) == B_OK) {
+
+			// Creating a menu of that font family
+			fontMenu = new BMenu (family);
+			fontMenu->SetRadioMode (true);	// I can set only one item as "in use"
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Don't comment the obvious. Below too.

File: DeskNoteView.cpp [hunk 7]
```diff
@@ -320,10 +427,14 @@ DeskNoteView::_ShowContextMenu(BPoint where)
 	bool canEdit = textView->IsEditable();
 	int32 length = textView->TextLength();
 
-	BPopUpMenu* menu = new BPopUpMenu(B_EMPTY_STRING, false, false);
+	BPopUpMenu* menu = new BPopUpMenu(B_EMPTY_STRING, false, false,B_ITEMS_IN_COLUMN);
 
 	colorMenu = new BMenu(B_TRANSLATE("Color"), 0, 0);
 	_BuildColorMenu(colorMenu);
+	
+	styleMenu = new BMenu(B_TRANSLATE("Font Style"));
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Sentence casing. Though just "Font" would be enough, I suppose.

File: DeskNoteView.cpp [hunk 8]
```diff
@@ -370,6 +484,41 @@ DeskNoteView::SaveNote(BMessage* msg)
...
+
+	font.SetFamilyAndStyle (fontFamily, fontStyle);
+	textView->SetFontAndColor(&font);
+
+	superItem = styleMenu->FindItem (fontFamily);
+	
+		if (superItem != NULL )
+			superItem->SetMarked (true);	// Check the one that was selected
+
+	menuItem = styleMenu->FindItem("Black");
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Why hardcode "Black" to checkmark?

**Review by humdingerb — CHANGES_REQUESTED** [review]
* Please use the haiku coding style, as does the rest of the code. There's haiku-format in HaikuDepot that can help with that. I commented a few things, but there are more violations. Also remove additionally inserted newlines.
* The current font menu items aren't checkmarked.
* The replicated view doesn't show the text with the set font.
* Please squash related commits before submitting a PR.

**Comment by BarmenduC** [comment]
> * Please use the haiku coding style, as does the rest of the code. There's haiku-format in HaikuDepot that can help with that. I commented a few things, but there are more violations. Also remove additionally inserted newlines.
> * The current font menu items aren't checkmarked.
> * The replicated view doesn't show the text with the set font.
> * Please squash related commits before submitting a PR.

Before performing the code-style cleanup, I'll read up the haiku coding style completely. Would perform the clean up soon enough.

Just to make sure, the tab style for variable declaration is to be dropped everywhere (even in header files) or just inside this file?

The stuff that I mentioned that is still remaining to be fixed are namely these two - 

1. Changed Font not appearing in replicants
2. Checkmark not appearing in-front of selected font

I had planned to keep working upon these issues after this PR gets merged, and to mention these 2 as separate issues so that, in case I am incapable of making it work, someone else can take up the work later. Would that be okay?

**Comment by humdingerb** [comment]
> Just to make sure, the tab style for variable declaration is to be dropped everywhere (even in header files) or just inside this file?

Just in the cpp file you changed.

> 1. Changed Font not appearing in replicants
> 2. Checkmark not appearing in-front of selected font
>
> I had planned to keep working upon these issues after this PR gets merged, and to mention these 2 as separate issues so that, in case I am incapable of making it work, someone else can take up the work later. Would that be okay?

I suppose so.
BTW, I recommend getting familiar with using 'git' from the commandline and using feature branches (see https://github.com/haikuports/haikuports/wiki/DevelopmentModel for an explanation). It's much more powerful and even convenient, once you get used to it.

**Comment by BarmenduC** [comment]
Requested changes have been mostly fixed. Will be keeping my out for more code-style fixes while I keep working upon  this further.

**Review by humdingerb — CHANGES_REQUESTED** [review]
Please have another look at the coding style at https://www.haiku-os.org/development/coding-guidelines . As mentioned, "haiku-format" from HaikuDepot can help you.

* Curly brackets at the end of "switch" statements
* Same for "for" and "if" conditions
* Space after "for", "if" etc.
* Space around operators ( =, >= etc)

File: DeskNoteView.cpp [hunk 9]
```diff
@@ -162,13 +156,15 @@ DeskNoteView::MessageReceived(BMessage* msg)
 			aboutWin->AddDescription(B_TRANSLATE(
 				"Pin little notes as replicants on your Desktop.\n"
 				"Dropped colors change the background of a note."));
-			const char* extraCopyrights[] = {
+			const char* extraCopyrights[] =
+			{
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Curly bracket at the end of previous line.

File: DeskNoteView.cpp [hunk 10]
```diff
@@ -261,74 +259,74 @@ DeskNoteView::MouseDown(BPoint point)
...
-	font_family		plainFamily,
-					family;
-	font_style		plainStyle,
-					style;
-	BMenuItem*		menuItem = NULL;
-	int32			numFamilies,	
-					numStyles;
-	BMenu*			fontMenu;
-	uint32			flags;
+	font_family	plainFamily, family;
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Space insted of tab.

File: DeskNoteView.cpp [hunk 11]
```diff
@@ -558,24 +560,26 @@ DeskNoteView::ResizeViewMethod(void* data)
 	float x, y;
 	DeskNoteView* theView = (DeskNoteView*) data;
 
-	do {
+	do
+	{
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Curly bracket at the end of previous line.

**Review by humdingerb — CHANGES_REQUESTED** [review]

**Comment by BarmenduC** [comment]
Manual code-style-fixes performed and .haiku-format added

File: DeskNoteView.cpp [hunk 12]
```diff
@@ -156,15 +156,13 @@ DeskNoteView::MessageReceived(BMessage* msg)
 			aboutWin->AddDescription(B_TRANSLATE(
 				"Pin little notes as replicants on your Desktop.\n"
 				"Dropped colors change the background of a note."));
-			const char* extraCopyrights[] =
-			{
+			const char* extraCopyrights[] =
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Opening curly backet is missing.

File: DeskNoteView.cpp [hunk 13]
```diff
@@ -176,23 +174,25 @@ DeskNoteView::MessageReceived(BMessage* msg)
 			aboutWin->Show();
 			break;
 		}
+
```
Comment (humdingerb) [inline DeskNoteView.cpp]: Between every case-statement, there's now a tab. Please remove, no empty line needed between cases.

**Review by humdingerb — CHANGES_REQUESTED** [review]
Much better. :)
Please remove the haiku.format file. It's only needed somewhere in the folder tree when executing the haiku-format tool.
There are quite a few stray whitespaces around. I recommend installing Koder and "Highlight trailing whitespaces" in its settings. Then they are easily to spot. Maybe even try "Trim trailing whitespaces" from its "Edit" menu.

**Comment by BarmenduC** [comment]
Changes performed :-
1. Removed .haiku-format
2. Trimmed trailing whitespaces using Koder
3. Added the missing opening bracket
4. Removed newlines after case statements

BTW, I had a query -> How did the Haiku developers settle on this codestyle (If this is not the right place to ask stuff like this, do tell me where I can do that)(IRC maybe)

**Review by humdingerb — APPROVED** [review]

**Comment by humdingerb** [comment]
> BTW, I had a query -> How did the Haiku developers settle on this codestyle (If this is not the right place to ask stuff like this, do tell me where I can do that)(IRC maybe)

Well, when many people work on a large project for a long time, there needs to be a style guide. The main developers agreed upon the rules very quickly and it changed only very little after that. While the HaikuArchives projects aren't strictly held to these style rules, it is a huge advantage to try have everyone adopt it. Makes it easy to contribute to any project without having to relearn a new style, or god-forbid,everyone doing their own thing and the code becoming a huge smorgasbord of styles...

I'll merge this for now. If it turns out you can't continue working on the open issues, and nobody else does, we can revert the changes.

In the future, please use a feature branch and squash your commits to sensible code bites.

**Pull request merged**